samcli/commands/deploy/deploy_context.py

```diff
@@ -264,14 +267,25 @@ def deploy(
                     if not click.confirm(f"{self.MSG_CONFIRM_CHANGESET}", default=False):
                         return
 
-                self.deployer.execute_changeset(result["Id"], stack_name, disable_rollback)
-                self.deployer.wait_for_execute(stack_name, changeset_type, disable_rollback)
+                # Stop the rollback in the case of DO_NOTHING or if this is a new stack and DELETE is specified
+                do_disable_rollback = (
```

Comment:
Can we have a comment here explaining that DO_NOTHING is exactly the same thing as disable rollback? Quoting from CFN documentation:

> To preserve successfully provisioned resources during a stack create operation
Specify the disable-rollback option or on-failure DO_NOTHING enumeration during a [create-stack](https://docs.aws.amazon.com/cli/latest/reference/cloudformation/create-stack.html) operation.